src/Finder/SourceFilesFinder.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright © 2017 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+declare(strict_types=1);
+
+namespace Infection\Finder;
+
+use Symfony\Component\Finder\Finder;
+
+class SourceFilesFinder
+{
+    private $sourceDirectories;
+    private $excludeDirectories;
+
+    /**
+     * @param string[] $sourceDirectories
+     * @param string[] $excludeDirectories
+     */
+    public function __construct(array $sourceDirectories, array $excludeDirectories)
+    {
+        $this->sourceDirectories = $sourceDirectories;
+        $this->excludeDirectories = $excludeDirectories;
+    }
+
+    public function getSourceFiles(string $filter = ''): Finder
+    {
+        $finder = new Finder();
+
+        array_walk($this->excludeDirectories, function ($excludePath) use ($finder) {
+            $finder->notPath($excludePath);
+        });
+
+        if ('' === $filter) {
+            $finder->in($this->sourceDirectories)->files();
+
+            return $finder;
+        }
+
+        $finder->in('.')->files();
```

Comment:
This is the last thing I would like to discuss (and probably update in PR). In master, we start to look from `sourceDirectories`, not from `.`. There are many libs where real sources are located much deeper than in the root. Example: [PHP-Parser](https://github.com/nikic/PHP-Parser/tree/master/lib/PhpParser), sources are in `./lib/PHP-Parser`. So relative path in master branch would start from `./lib/PHP-Parser`, in this PR it starts from `.`.

I made such changes to make it like in master.

```php
public function getSourceFiles(string $filter = ''): Finder
    {
        $finder = new Finder();

        array_walk($this->excludeDirectories, function ($excludePath) use ($finder) {
            $finder->notPath($excludePath);
        });

        $finder->in($this->sourceDirectories)->files();

        $filters = explode(',', $filter);
        array_walk($filters, function ($fileFilter) use ($finder) {
            $finder->path($fileFilter);
        });

        return $finder;
    }
```

Does it make sense? :) Do you think relative path should start from `.` instead of what is configured as a `src` in `infection.json.dist`?